content/scripts/build.js

```diff
@@ -481,41 +488,41 @@ class Builder {
     // that every 'en-US' document that has been translated, will have a list
     // of other locales and slugs.
     let countBrokenTranslationOfDocuments = 0;
-    Object.values(this.allTitles)
-      .filter((data) => data.translation_of)
-      .forEach((data) => {
-        const parentURL = buildMDNUrl("en-US", data.translation_of);
-        const parentData = this.allTitles[parentURL];
-
-        // TODO: Our dumper is not perfect yet. We still get bad
-        // 'translation_of' references.
-        // I.e. a localized document's 'index.yaml' says its
-        // 'translation_of' is 'Web/Foo/Bar' but there is actually no en-US
-        // document by that slug!
-        // We're working on it in the dumper and this problem is known also
-        // in the 'ensureAllTitles()' method which at least debug logs the
-        // bad ones and warn logs about a total count of bad documents.
-        // Once the dumper has matured, we'll remove this defensive style and
-        // throw an error here. That'll be a form of validation-by-building
-        // which can really help our CI trap content edit PRs that sets
-        // or gets these references wrong.
-
-        if (parentData) {
-          if (!parentData.hasOwnProperty("translations")) {
-            parentData.translations = [];
-          }
-          parentData.translations.push({
-            locale: data.locale,
-            slug: data.slug,
-          });
-        } else {
-          countBrokenTranslationOfDocuments++;
-          this.logger.debug(
-            `${data.locale}/${data.slug} (${data.file}) refers to a ` +
-              "en-US translation_of that doesn't exist."
-          );
+    this.allTitles.forEach((data) => {
+      if (!data.translation_of) return;
```

Comment:
Perhaps I'm a hipster but I feels like this looks less cool than the `.filter()` but `Array.prototype.filter` isn't a stream, is it?
Anyway, this line is the only way I know of how to iterate on a `Map` instance and combining it with a filter.

Reply:
Yours is good too, but what do you think of the following, which I find a tiny bit more readable (but that's just my taste -- although perhaps there is a slight performance gain in not calling a function -- the arrow function -- on every iteration?):
```javascript
for (const data of this.allTitles.values()) {
  if (!data.translation_of) continue;
  ...
}
```
The `.values()` just returns an iterator over the map's values, so it doesn't cost anything.

Reply:
To be honest, I'm not sure which is "best". It's pretty clear that you generally prefer `for` loops and I generally prefer things like `.forEach` and `.filter` and those guys. That impression you have that "find a tiny bit more readable" is an important argument. 

For the record, both `this.allTitles.forEach(data => ...)` and `for (const data of this.allTitles.values()) {...` seems to have the exact performance characteristics. It's just moving a pointer in an "array" without making a new array, in memory, like you'd get with `Object.values(someObject)`. So then it's down to style. No?

Also for the record, suppose your logic is to exit the iterator as soon as you've found your needle, I think only the `for (const data of this.allTitles.values()) {` can do that. E.g. 
```javascript
for (const item of things) {
  if (item === 42) {
     return "found it!"
  }
}
```
which is `O(log n)`.
On the flipside if you need to have an index, you get that for free with `.forEach`. 
```javascript
let i = 0
for (const item of things) {
  if (i === things.length - 1) {
    console.log("Last one is", item)
  }
  i++;
}

// vs. 
things.forEach((item, i) => {
  if (i === things.length - 1) {
    console.log("Last one is", item)
  }
});
```

An advantage for the composable ones like `.forEach` is that they allow you to compose multiple together. For example, 
```javascript
const needles = things
  .filter(filterA)
  .filter(filterB)
  .filter(filterZ)
  .map(transformA)
  .map(transformB)
  .map(transformZ)
```
which can make each of those little callbacky functions feel independent and self contained  instead of all being bundled with notation all in one big block like:
```javascript
const needles = []
for (thing of things) {
  if (filterA(thing) && filterB(thing) && filterZ(thing) {
    needles.push(transformA(transformB(transformZ(thing))))
  }
}
```
But I'm not sure either is much prettier than the other. 

I'm struggling to take sides on this one. 
I know that my Python is mixed with a bit of both. `for` loops mostly unless it's easy to express with list comprehension. E.g.

```python
def process_needles(stuff):
    for thing in [x.strip() for x in stuff if x.strip() and not x.startswith('#')]:
         if thing.foo and not thing.bar:
             print("dbg:", thing)
             do_something()
```

So I'm not sure what's best. It certainly would be nice to have some consistency throughout the code. **With that in mind, I'm happy to change this line**. 

Perhaps the takeaway is to use either when it doesn't matter and use `for` loop when you need to break early and use `.forEach` when you need an index counter.